core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -453,14 +454,21 @@ final class ShuffleBlockFetcherIterator(
             input = streamWrapper(blockId, in)
             // Only copy the stream if it's wrapped by compression or encryption, also the size of
             // block is small (the decompressed block is smaller than maxBytesInFlight)
-            if (detectCorrupt && !input.eq(in) && size < maxBytesInFlight / 3) {
+            if (detectCorrupt && !input.eq(in)) {
               isStreamCopied = true
+              streamCompressedOrEncrypted = true
```

Comment:
I wonder if we should do the inputStream checks even if `detectCorrupt` is false, and perhaps even change that default.  Or add another flag just for doing this memory copy, as its more expensive.

Also I think the comments around here need to be updated to explain what is happening now.

Reply:
Sure, I can change it to always detect corruption. Which default did you mean though, the default value of `detectCorrupt` is true.